Source/Scene/Camera.js

```diff
@@ -3230,9 +3240,35 @@ define([
         Matrix4.clone(camera._transform, result.transform);
         result._transformChanged = true;
 
+        if (defined(camera.cameraLimiter)) {
+            result.cameraLimiter = CameraLimiter.clone(camera.cameraLimiter, result.cameraLimiter);
+        }
+
         return result;
     };
 
+    var saveOrientation = new HeadingPitchRoll();
+    function limitPosition(camera) {
+        // following the convention from _adjustHeightForTerrain to maintain correct orientation
+        // before and after position is changed
+        if (defined(camera.cameraLimiter) && camera._mode === SceneMode.SCENE3D) {
+            // store camera's hpr values
+            saveOrientation.heading = camera.heading;
+            saveOrientation.pitch = camera.pitch;
+            saveOrientation.roll = camera.roll;
+
+            // update position
+            camera.position = camera.cameraLimiter.limitPosition(camera.position, camera.position);
```

Comment:
Don't set the `camera.position` here.  Instead, just send the result to `setView`.  `setView` has logic that will make it work in all 3 scene modes